### PR TITLE
Fixed being able to identify falsewalls by right clicking them

### LINKED ISSUE #30512: You can just right-click to check for False Walls
image

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #30512

:cl:
* bugfix: Fixed being able to identify falsewalls by right clicking them.
* bugfix: Fixed false reinforced walls not having the same description as reinforced walls.
* rscadd: False walls now give an extra message on examination when standing adjacent to them.

### CODE REVIEW

**Comment by Exxion** [comment]
OK what about mesons just seeing void

**Comment by DeityLink** [comment]
> 
> 
> OK what about mesons just seeing void

They will only see void if the falsewalls are open. They show up on mesons.

**Comment by DeityLink** [comment]
Actually they should show on mesons now that you mention it since otherwise you can already identify falsewalls this way 
image

**Comment by Exxion** [comment]
You can see the floor with mesons even if it has `invisibility = 101`? That's very odd.

**Comment by DeityLink** [comment]
> 
> 
> You can see the floor with mesons even if it has `invisibility = 101`? That's very odd.

nah, that's another problem I'm bringing up. They'll just see void.

**Comment by DeityLink** [comment]
This doesn't remove the typo when examining them but I think that one is at least a bit more thematic, compared to the completely client-side act of right clicking.

**Comment by DeityLink** [comment]
By @DamianX 's request I've also fixed the typos.

I changed the fix so it removes mouse_opacity instead of making invisibility = 101, so you will not see a black void while wearing mesons (though you can still use mesons and notice the wrong wall smoothing, and that you can't right click a floor)

**Comment by Exxion** [comment]
>I've also fixed the typos.

Boo

**Comment by boy2mantwicethefam** [comment]
Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

**Comment by DeityLink** [comment]
> 
> 
> Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

Tell you what I'm gonna try later. I think I see how I could do that.

**Comment by Kurfursten** [comment]
I thought the different wall description was a feature. Just look at it closely.

**Comment by DeityLink** [comment]
> 
> 
> I thought the different wall description was a feature. Just look at it closely.

I thought as well but then we noticed the typo was also in the regular r_wall description. So we deduced that the typo was originally everywhere, and only fixed properly in regular walls.

**Comment by DeityLink** [comment]
> 
> 
> Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

Victory


https://user-images.githubusercontent.com/7573912/132132978-57be9c60-2010-4e2f-81c5-44992d299379.mp4

**Comment by BomberBro** [comment]
I think the typo is fine, since when were false walls supposed to be perfect facsimiles of real walls?

**Comment by DeityLink** [comment]
I'd rather that instead of a typo you get an extra message on examine() when adjacent

File: code/modules/clothing/glasses/scanners.dm [hunk 1]
```diff
@@ -158,6 +161,41 @@ var/list/meson_wearers = list()
...
+	update_mob(M)
+	..()
+
+/obj/item/clothing/glasses/scanner/meson/proc/update_mob(var/mob/new_mob)
+	if (new_mob == viewing)
+		clear()
+		apply()
+		return
+
+	if (new_mob != viewing)
```
Comment (DamianX) [inline code/modules/clothing/glasses/scanners.dm]: isn't this always true?

Comment (DeityLink) [reply code/modules/clothing/glasses/scanners.dm]: Nope. The proc is called with a null arg to clear() and remove the viewer. Same as with the material scanner below.

**Comment by BomberBro** [comment]
> I'd rather that instead of a typo you get an extra message on examine() when adjacent

That sounds good too, if false walls had any real effort put into them I'd be all for making them extremely sneaky, but as it stands they take about 2 seconds to set up

File: code/modules/clothing/glasses/scanners.dm (hunk 1 above)
Comment (DamianX) [reply code/modules/clothing/glasses/scanners.dm]: The `if` before this one checks whether `new_mob` is equal to `viewing`. If it is, it returns. So the only way to reach this second `if`, is for `new_mob` to not be equal to `viewing`.

Comment (DeityLink) [reply code/modules/clothing/glasses/scanners.dm]: oh, well then I fix it in both places

**Comment by DeityLink** [comment]
> 
> 
> > I'd rather that instead of a typo you get an extra message on examine() when adjacent
> 
> That sounds good too, if false walls had any real effort put into them I'd be all for making them extremely sneaky, but as it stands they take about 2 seconds to set up

To be fair as long as a light push is all it takes top open them, they really shouldn't take much more effort.

**Comment by SonixApache** [comment]
not being able to see through fake walls on mesons is something that should be noted in the :cl:

**Comment by TGFH** [comment]
Why shouldn't false walls be identifiable with mesons? It would be stupid if literally the only way to identify them is to spam click walls.

@DeityLink 
>To be fair as long as a light push is all it takes top open them, they really shouldn't take much more effort

People get very upset about random wallclicking because it's basically just metagaming.

**Comment by Exxion** [comment]
Yeah this PR is really sprawling away from its original purpose

**Comment by SonixApache** [comment]
Maybe split off the meson thing to its own PR? I for one would like it, it'd help maint hiders actually hide in maint.

**Comment by Exxion** [comment]
Meson thing should definitely be its own PR. The rest of it is probably related enough to stay as one PR even if I'd rather it were two.

**Comment by DeityLink** [comment]
> 
> People get very upset about random wallclicking because it's basically just metagaming.

Yeah my point is that metagamers shouldn't be able to hide so easily obvious attempts at metagaming. (I'd rather see metagamers try and push or examine every wall than allowing them to right click them at any range just to check.)

**Comment by DeityLink** [comment]
Meson changes split off to #30598

**Comment by TGFH** [comment]
> 
> 
> > People get very upset about random wallclicking because it's basically just metagaming.
> 
> Yeah my point is that metagamers shouldn't be able to hide so easily obvious attempts at metagaming. (I'd rather see metagamers try and push or examine every wall than allowing them to right click them at any range just to check.)

My gripe was the mesons change, not the right clicking thing.

**Pull request merged**